include/circt/Support/Naming.h

```diff
@@ -0,0 +1,30 @@
+//===- Naming.h - Utilities for handling names ------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef CIRCT_SUPPORT_NAMING_H
+#define CIRCT_SUPPORT_NAMING_H
+
+#include "circt/Support/LLVM.h"
+
+namespace circt {
+
+/// Return true if this is a possibly useless temporary name.
+bool isUselessName(StringRef name);
+
+/// Choose a good name for an item from two options.
+StringRef chooseName(StringRef a, StringRef b);
+
+/// Choose a good name for an item from two options.
+StringAttr chooseName(StringAttr a, StringAttr b);
+
+/// Choose the better name between two ops.
+StringAttr chooseName(Operation *a, Operation *b);
```

Comment:
It may be worth mentioning this assumes Operation uses `sv.namehint` to store the "name" which is not necessarily true for all Dialects in CIRCT (e.g., FIRRTL).